frontend/src/Components/Login.js

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector, batch } from "react-redux";
+import { useNavigate, Link } from "react-router-dom";
+import { API_URL } from "utils/utils";
+import user from "reducers/user";
+import styled from "styled-components";
+
+const OuterWrapper = styled.div`
+padding: 20px 0 0 0;
+`;
+
+const InnerWrapper = styled.div`
+  // max-width: 940px;
+  width: 100%;
+  margin: 0 auto;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+`;
+
+const Radiobuttons = styled.div`
+display: flex;
+flex-direction: row;
+margin: 15px;
+`
+
+const InputWrapper = styled.div`
+display: flex;
+flex-direction: column;
+margin: 15px;
+`
+
+const InputForm = styled.input`
+padding: 10px;
+display: flex;
+flex-direction: column;
+`;
+
+const Login = () => {
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [mode, setMode] = useState("login");
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const accessToken = useSelector((store) => store.user.accessToken);
+  useEffect(() => {
+    if (accessToken) {
+      navigate("/");
+    }
+  }, [accessToken]);
+
+  const onFormSubmit = (event) => {
+    event.preventDefault();
+    const options = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({ username: username, password: password }),
+    };
+    fetch(API_URL(mode), options)
+      .then((response) => response.json())
+      .then((data) => {
+        if (data.success) {
+          batch(() => {
+            dispatch(user.actions.setUsername(data.response.username));
+            dispatch(user.actions.setUserId(data.response.id));
+            dispatch(user.actions.setAccessToken(data.response.accessToken));
+            dispatch(user.actions.setError(null));
+          });
+        } else {
+          alert("wrong credentials");
```

Comment:
Nice solution with the alert-sign! We didn't know how to display the error message, but we really like this solution!